src/ContainerCheckpointModal.jsx

```diff
@@ -1,89 +1,70 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Button, Checkbox, Form, Modal } from '@patternfly/react-core';
-import { DialogsContext } from "dialogs.jsx";
+import { useDialogs } from "dialogs.jsx";
 import cockpit from 'cockpit';
 
 import * as client from './client.js';
 
 const _ = cockpit.gettext;
 
-class ContainerCheckpointModal extends React.Component {
-    static contextType = DialogsContext;
+const ContainerCheckpointModal = ({ containerWillCheckpoint, onAddNotification }) => {
+    const Dialogs = useDialogs();
+    const [inProgress, setProgress] = useState(false);
+    const [keep, setKeep] = useState(false);
+    const [leaveRunning, setLeaveRunning] = useState(false);
+    const [tcpEstablished, setTcpEstablished] = useState(false);
+    const [ignoreRootFS, setIgnoreRootFS] = useState(false);
 
-    constructor(props) {
-        super(props);
-        this.state = {
-            inProgress: false,
-            keep: false,
-            leaveRunning: false,
-            tcpEstablished: false,
-            ignoreRootFS: false
-        };
-        this.handleChange = this.handleChange.bind(this);
-    }
-
-    handleChange(checked, event) {
-        if (event.target.type === "checkbox")
-            this.setState({ [event.target.name]: event.target.checked });
-    }
-
-    handleCheckpointContainer(args) {
-        const Dialogs = this.context;
-        const container = this.props.containerWillCheckpoint;
-        this.setState({ inProgress: true });
-        client.postContainer(container.isSystem, "checkpoint", container.Id, args)
+    const handleCheckpointContainer = () => {
+        setProgress(true);
+        client.postContainer(containerWillCheckpoint.isSystem, "checkpoint", containerWillCheckpoint.Id, {
+            keep,
+            leaveRunning,
+            tcpEstablished,
+            ignoreRootFS,
+        })
                 .catch(ex => {
-                    const error = cockpit.format(_("Failed to checkpoint container $0"), container.Names);
-                    this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
-                    this.setState({ inProgress: false });
+                    const error = cockpit.format(_("Failed to checkpoint container $0"), containerWillCheckpoint.Names);
+                    onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                    setProgress(false);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1146-20221125-083242-f4438564-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)